modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -91,6 +91,23 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     });
   };
 
+  const toggleCaption = () => {
+    getSelectionStartCellOrCaption(editor).each((cellOrCaption) => {
+      TableLookup.table(cellOrCaption, isRoot).each((table) => {
+        SelectorFind.child(table, 'caption').fold(
+          () => {
+            const caption = SugarElement.fromTag('caption');
+            Insert.append(caption, SugarElement.fromText('Caption'));
+            Insert.appendAt(table, caption, 0);
+          },
+          Remove.remove
```

Comment:
Discussed elsewhere but reminder: cursor/selection goes somewhere weird after this because we've removed the element it was in, so we should purposefully put it somewhere. I reckon start of the first cell works.

Reply:
- [ ] fix

Reply:
I suspect this was actually a bug with the dialog before, as it also deleted without updating the selection :thinking: One thing to be careful of here, is that we only should move the selection if it was within the caption.